### PR TITLE
Fix #2252 (consistent typing for the 'delay' log property)

### LINKED ISSUE #2252: Inconsistent format of delay field in JSON logs
When I re-deployed the ELK stack for Nimbus fleet I found a lot of errors like this in Logstash:
```
 Could not index event to Elasticsearch.
{
  :status=>400,
  :action=>["index", {:_id=>nil, :_index=>"logstash-2021.01.20", :routing=>nil, :_type=>"_doc"},#<LogStash::Event:0x75d7912e>], 
      :response=>{
        "index"=>{
          "_index"=>"logstash-2021.01.20",
          "_type"=>"_doc",
          "_id"=>"YbBGIXcBUiN4-Z-XYxJ8",
          "status"=>400, 
          "error"=>{"type"=>"mapper_parsing_exception",
          "reason"=>"failed to parse field [json.delay] of type [text] in document with id 'YbBGIXcBUiN4-Z-XYxJ8'. Preview of field's value: '{value=1667882472}'",
          "caused_by"=>{"type"=>"illegal_state_exception", 
          "reason"=>"Can't get text on a START_OBJECT at 1:800"
        }
      }
    }
  }
}
```
This happens because of the value of field `delay` is sometimes and object and sometimes a string. Examples:
```
{"lvl":"DBG","ts":"2021-01-20 19:28:14.263+00:00","msg":"Attestation received","tid":1,"file":"eth2_processor.nim:322","delay":{"value":7263939127},"wallSlot":455840,"attestation":{"aggregation_bits":"0x0000000000000001000000000000000002","data":{"slot":455840,"index":3,"beacon_block_root":"2d0dd34e","source":{"epoch":14244,"root":"48596473"},"target":{"epoch":14245,"root":"2d0dd34e"}},"signature":"84ac0bde"},"committeeIndex":3}
```
```
{"lvl":"NOT","ts":"2021-01-20 19:28:14.164+00:00","msg":"Attestation sent","topics":"beacval","tid":1,"file":"validator_duties.nim:220","attestation":{"aggregation_bits":"0x0000000000000200000000000000000002","data":{"slot":455840,"index":3,"beacon_block_root":"2d0dd34e","source":{"epoch":14244,"root":"48596473"},"target":{"epoch":14245,"root":"2d0dd34e"}},"signature":"9763d1de"},"validator":"a233767b","delay":"3s164ms948us409ns","indexInCommittee":49}
```
As seen here in one case it's `"delay":"3s164ms948us409ns"` and in another it's `"delay":{"value":7263939127}` which makes it impossible to consistently parse by Logstash and index in Elasticsearch. At least not without some very weird edge case rules.

### PR DESCRIPTION
**Pull request by zah**

### CODE REVIEW

File: tests/test_timeutils.nim [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# beacon_chain
+# Copyright (c) 2018 Status Research & Development GmbH
```
Comment (tersec) [inline tests/test_timeutils.nim:2]: 2018?

**Review by tersec — APPROVED** [review]

File: beacon_chain/time.nim [hunk 2]
```diff
@@ -104,11 +104,88 @@ proc addTimer*(fromNow: Duration, cb: CallbackFunc, udata: pointer = nil) =
...
+
+  let hours = v div nanoseconds(Hour)
+  if hours > 0:
+    if hours >= 2:
+      result = $hours & " hours"
+    else:
+      result = "1 hour"
+    v = v mod nanoseconds(Hour)
+
+  let minutes = v div nanoseconds(Minute)
```
Comment (tersec) [inline beacon_chain/time.nim:126]: This pivot from `return` to `result` was quite the plot twist.

File: tests/test_timeutils.nim [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  timedTest "humaneStr":
+    check:
+      humaneStr(2.weeks) == "2 weeks"
+      humaneStr(2.weeks + 2.days) == "2 weeks"
+      humaneStr(2.weeks + 7.days) == "3 weeks"
+      humaneStr(1.weeks + 3.days) == "10 days"
+      humaneStr(1.weeks + 3.days + 2.hours) == "10 days"
+      humaneStr(5.days + 3.hours) == "5 days"
+      humaneStr(1.days + 3.hours + 10.minutes) == "27 hours 10 minutes"
+      humaneStr(1.days + 15.seconds + 342.milliseconds) == "24 hours 15 seconds"
```
Comment (arnetheduck) [inline tests/test_timeutils.nim:25]: hm, rounding? not sure that's a good idea

File: tests/test_timeutils.nim [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import
+  unittest, ./testutil,
+  chronos/timer,
+  ../beacon_chain/time
+
+suiteReport "Time utilities":
+  timedTest "humaneStr":
+    check:
+      humaneStr(2.weeks) == "2 weeks"
+      humaneStr(2.weeks + 2.days) == "2 weeks"
```
Comment (arnetheduck) [inline tests/test_timeutils.nim:19]: this rounding is quite overly aggressive as well - ie these differences are significant in our usage of time in the logs

**Comment by arnetheduck** [comment]
I'd generally say that for the purpose of logging (as opposed to a dialog box you show to a less interested user), the original format of `3s164ms948us409ns` is both more precise and more efficient to work with - the loss of precision makes using the logs for debugging a lot harder because suddenly you have to consider the floor rounding.

**Comment by zah** [comment]
Most actual values in the logs will appear with millisecond precision (e.g. 1.231 seconds). Can you suggest specific examples where this wouldn't be enough to reason about timing? I guess it really depends on who we consider the intended audience for the logs. If we insist that the logs are primarily for debugging, then yes, it may make more sense to use the precise, but harder to decipher value. If instead we say that they are part of the UI, then more care towards what goes onto the screen is appropriate.

**Comment by arnetheduck** [comment]
well, I often compare timestamps and look at processing in detail -- this has helped me decipher several, otherwise difficult-to-diagnose async races - ie the marginal difference between two rounds of processing will usually be interesting if a larger part of the processing is fixed, and that completely goes away with rounding. the precise value of the nanosecond can also often be traced or looked for to correlate with other logs which is another small "trick" - or a core dump to find it in memory etc etc.

these logs are primarily an audit trail for what has happened in the application - in some sense, they double as a poor man's UX as well, but timing is really important to eth2 and having imprecise timing logs seems counterproductive.

**Comment by arnetheduck** [comment]
ie this is also why we have strucutured logs: they're optimized for processing, not human reading - ie to make a diagram of block processing times for example.

**Comment by arnetheduck** [comment]
ping.. these new timestamps are making debugging harder - can we get the old ones back?

**Comment by zah** [comment]
OK, there you go:
https://github.com/status-im/nimbus-eth2/pull/2269
